modular_zubbers/code/modules/modular_computers/small_painting_printer.dm

```diff
@@ -0,0 +1,78 @@
+/obj/item/modular_computer/mini_painting_printer
+	name = "instant portrait printer"
+	desc = "Allows users to quickly print off desired artworks."
+	icon = 'icons/obj/devices/artefacts.dmi'
+	icon_state = "prototype9"
+	density = FALSE
+	starting_programs = list(
+		/datum/computer_file/program/portrait_printer/mini
+	)
+	internal_cell = /obj/item/stock_parts/power_store/cell/crap
+	stored_paper = 10
+	max_paper = 10
+	max_capacity = 1
+	max_idle_programs = 1
+	base_active_power_usage = 0.1 WATTS
+	base_idle_power_usage = 0.1 WATTS
+
+/obj/item/modular_computer/mini_painting_printer/Initialize(mapload)
+	. = ..()
+	var/datum/computer_file/program/portrait_printer/mini/printer_prog = find_file_by_name("portraitprinter_mini")
+	active_program = printer_prog
+
+/datum/computer_file/program/portrait_printer/mini
+	filename = "portraitprinter_mini"
+	filedesc = "Marlowe Treeby's Art Sharer"
+	size = 1
+	program_open_overlay = "dummy"
+	extended_desc = "Uses bluespace technology to print off a painting."
+	can_run_on_flags = PROGRAM_PDA
+	program_flags = PROGRAM_REQUIRES_NTNET
+	power_cell_use = 0 WATTS
+
+/datum/computer_file/program/portrait_printer/mini/print_painting(selected_painting)
+	. = ..()
+	var/datum/painting/chosen_portrait = locate(selected_painting) in SSpersistent_paintings.paintings
+	var/obj/item/wallframe/painting/frame
+	if(is_valid_frame(chosen_portrait) == TRUE)
```

Comment:
you can omit the == TRUE here
also, why is the else case here checking this again? that seems super wierd